pkg/apis/serving/v1alpha1/knativeserving_types.go

```diff
@@ -51,6 +51,15 @@ type IstioGatewayOverride struct {
 	Selector map[string]string `json:"selector,omitempty"`
 }
 
+// DisabledComponents skips certain sections of knative serving
+type DisabledComponents struct {
```

Comment:
I'm not against this being expressable in the operator's API at all, but I'm not sure this is the best API construction to use:

- The API conventions [advise against](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types) using booleans as a type for API fields
- It feels a little weird to have `spec.disabledComponents.certmanager = true` in the API in terms of polarity. I'd rather indicate disablement through a "false" sounding value.

On an entirely different note, unless we make some other change, I do not believe there is a way to "disable" use of cert-manager, unless you disable the use of admission webhooks, since cert-manager is used to mint the serving certificate from the webhook controller.